blog/content/post/2022-01-03-pcie_screamer.md

```diff
@@ -0,0 +1,218 @@
+   git clone git@github.com:3mdeb/pcie_screamer.git
+   cd pcie_screamer
+   git checkout 01aee32f54f15163fac870cda3739ddd77b3c03c
+   ```
+
+2. Prepare [enjoy-digital/litex](https://github.com/enjoy-digital/litex)
+   environment
+
+   ```
+   wget https://raw.githubusercontent.com/enjoy-digital/litex/master/litex_setup.py
+   chmod +x litex_setup.py
+   ./litex_setup.py --init --install --user
+   ```
+
+   > This step will extract plenty of files in the directory, be sure to execute
+   > it from the `pcie_screamer` project directory.
+
+3. Now, the building of `Xilinx BIT data` is working
+
+   ```
+   ./pcie_screamer.py --build
+   ```
+
+4. However, the loading fails
+
+   ```
+   $ ./pcie_screamer.py --load
+
+   <output truncated>
+
+   failed loading file build/gateware/top.bit to pld device 0
+   Traceback (most recent call last):
+     File "./pcie_screamer.py", line 160, in <module>
+       main()
+     File "./pcie_screamer.py", line 150, in main
+       prog.load_bitstream("build/gateware/top.bit")
+     File "/home/ibagnucki/git/pcie_screamer/litex/litex/litex/build/openocd.py", line 27, in load_bitstream
+       self.call(["openocd", "-f", config, "-c", script])
+     File "/home/ibagnucki/git/pcie_screamer/litex/litex/litex/build/generic_programmer.py", line 100, in call
+       raise OSError(msg)
+   OSError: Error occured during OpenOCD's call, please check:
+   - OpenOCD installation.
+   - access permissions.
+   - hardware and cable.
+   ```
+
+## Power problem
+
+Programming of the PCIeScreamer in the previous step failed. There could be
+several reasons why this could happen. It could be a broken programmer,
+misconfigured environment, or nonfunctional FPGA. The third option the problem
+with the FPGA seem to be most probable as
+[LambdaConcept's instructions](https://docs.lambdaconcept.com/screamer/getting_started.html#boot-the-target-system)
+warn about correct power delivery to the board and explain that the green diode
+LD3 should be on.
+
+The diode is indeed on, but only if JTAGSerial Programmer is connected to the
+power too. When the Programmer is disconnected, the led goes dark too. This LED
+behavior may (but not necessarily have to) indicate invalid power of the board.
+Another indication that something may be wrong with the PCIe connection is that
+no new PCIe device was shown on the tested machine using the `lspci` command.
+
+![](/img/PCIeScreamer_LED.jpg)
+
+The incorrect power delivered to the FPGA could be the result of the faulty PCIe
+low-profile riser card used to connect the PCIeScreamer to the motherboard.
+
+![](/img/PCIeScreamer_Adapter.jpg)
+
+
+## Perspectives on further work
+
+The next step to make PCIe testing work should be to check why programming
+FPGA didn't work, and if it is correctly connected.
+This work should include checking the FPGA with a different motherboard,
+preferably without a low-profile riser card. If that doesn't help, another JTAG
+programmer may solve the problem.
```

Comment:
In my opinion we should try again to upload configuration file into PCIScreamer using tested JTAG programmer and Viavado tools for programming FPGA. There is also alternative to synthesize project from sources in "System Verilog" and using Vivado as major toll for this purpose.